code/datums/votes/map_vote.dm

```diff
@@ -0,0 +1,80 @@
+/datum/vote/map_vote
+	name = "Map"
+
+/datum/vote/map_vote/New()
+	. = ..()
+
+	default_choices = list()
+
+	// Fill in our default choices with all of the maps in our map config, if they are votable and not blocked.
+	var/list/maps = shuffle(global.config.maplist)
+	for(var/map in maps)
+		var/datum/map_config/possible_config = config.maplist[map]
+		if(!possible_config.votable || (possible_config.map_name in SSpersistence.blocked_maps))
+			continue
+
+		default_choices += possible_config.map_name
+
+/datum/vote/map_vote/create_vote()
+	. = ..()
+
+	// Before we create a vote, remove all maps from our choices that are outside of our population range.
+	// Note that this can result in zero remaining choices for our vote, which is not ideal (but technically fine).
+	for(var/map in choices)
+		var/datum/map_config/possible_config = config.maplist[map]
+		if(possible_config.config_min_users > 0 && GLOB.clients.len < possible_config.config_min_users)
+			choices -= map
+
+		else if(possible_config.config_max_users > 0 && GLOB.clients.len > possible_config.config_max_users)
+			choices -= map
+
+/datum/vote/map_vote/toggle_votable(mob/toggler)
+	if(!check_rights_for(toggler?.client, R_ADMIN))
```

Comment:
why would toggler not exist? are you sure this isn't suppressing a runtime?

Reply:
I intended the "`toggler?`" check as an anti-spoof check for someone calling it when they shouldn't - but, I suppose by extension it's also silencing a runtime